File: jetty-router/src/main/java/com/practicaldime/jesty/app/Sample.java
package com.practicaldime.jesty.app;

import java.util.function.BiFunction;

public class Sample {

	public String echo(String value) {
		return value;
	}

	public String echo(String value, int modifier) {
		switch(modifier) {
			case 0:{
				return value.toUpperCase();
			}
			case 1:{
				return new StringBuilder(value).reverse().toString();
			}
			default:
				return value;
		}
	}
	
	public String echo(String value, int modifier, BiFunction<String, Integer, String> action) {
		return action.apply(value, modifier);
	}
	
	public static void main(String...args) {
		System.out.println(new Sample().echo("Test from java", 1));
	}
}

File: jetty-router/www/sample.js
let Sample = Java.type('com.practicaldime.jesty.app.Sample');

let Date = Java.type('java.util.Date');
let DateFormat = Java.type('java.text.SimpleDateFormat');

function now() {
    return new DateFormat("hh:mm:ss a").format(new Date());
}

let app = new Sample();

let func = function(val, mod){
    switch(mod){
        case 0:
            return val.toUpperCase();
        case 1:
            return val.split("").reverse().join("");
        default:
            return now() + " : " + val;
    }
};

print(app.echo("Test App", -1, func));